pkg/apis/core/v1alpha1/types.go

```diff
@@ -186,3 +186,18 @@ type CDIList struct {
 	// Items provides a list of CDIs
 	Items []CDI `json:"items"`
 }
+
+//CDIUserConfig provides a user configuration for CDI
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type CDIUserConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec CDIUserConfigSpec `json:"spec"`
```

Comment:
Should have `Spec` and a `Status` members.  Maybe look something like this:

```
type CDIConfigSpec struct {
     UploadProxyURLOverride *string //if controller can't derive the URL or it's incorrect, user can override
}

type CDIConfigStatus struct {
     UploadProxyURL *string //this is the field users look at to find the upload proxy url.  Controller sets it to UploadProxyURLOverride if set, otherwise attempts to discover it
}
```